include/mlir-tcp/Dialect/IR/TcpBase.td

```diff
@@ -97,4 +59,12 @@ class Tcp_BinaryElementwiseOp<string mnemonic, list<Trait> traits = []> :
         SameOperandsAndResultShape])> {
 }
 
+//===----------------------------------------------------------------------===//
+// Tcp Types Base
+//===----------------------------------------------------------------------===//
+
+class Tcp_Type<string name, string typeMnemonic> : TypeDef<Tcp_Dialect, name> {
+    let mnemonic = typeMnemonic;
+}
+
```

Comment:
Why not move this also to `TcpTypes.td`?

Reply:
No particular reason other than maintaining the pattern we followed for `Tcp_Dialect` and `Tcp_Op`. Upstream dialects don't follow any particular pattern either - e.g. Tosa uses a separate `TosaOpsBase.td` and `TosaTypesBase.td` whereas GPU dialect uses the same `GPUBase.td` for dialect, enum, type, interfaces and attrs.